src/HomogeneousCalorimeter_geo.cpp

```diff
@@ -406,58 +406,114 @@ static std::tuple<int, int> add_disk(Detector& desc, Assembly& env, xml::Collect
-  Material ring_material = desc.material(getAttrOrDefault<std::string>(plm, _U(material), "StainlessSteel"));
-  // Material hole_material     = desc.material(getAttrOrDefault<std::string>(plm, _U(material), "Air"));
-  PolyhedraRegular solid_world(12, 0., r12min, calo_module_length);
-  Tube             solid_sub(0., rmin, calo_module_length / 2., phimin, phimax);
-  SubtractionSolid calo_subtract(solid_world, solid_sub, Position(0., 0., 0.));
-  Volume           env_vol(std::string(env.name()) + "_envelope", calo_subtract, ring_material);
-  Transform3D      tr_global = RotationZYX(15. * degree, 0., 0.) * Translation3D(0., 0., 0.);
-  env_vol.setVisAttributes(desc.visAttributes(plm.attr<std::string>(_Unicode(vis_steel_gap))));
+  Material outer_ring_material     = desc.material(getAttrOrDefault<std::string>(plm, _U(material), "StainlessSteel"));
+  Material inner_ring_material     = desc.material(getAttrOrDefault<std::string>(plm, _U(material), "Copper"));
+  // Material hole_material     = desc.material(getAttrOrDefault<std::string>(plm, _U(material), "Vacuum"));
+
+
 
+
+  //==============================
   // Outer supporting frame
-  //
+  //==============================
+
   PolyhedraRegular solid_ring12(12, r12min, r12max, structure_frame_length);
-  Volume           ring12_vol("ring12", solid_ring12, ring_material);
-  Transform3D      tr_global_Oring = RotationZYX(15. * degree, 0., 0.) * Translation3D(0., 0., -20. * cm);
+  Volume ring12_vol("ring12", solid_ring12, outer_ring_material);
+  Transform3D tr_global_Oring = RotationZYX(15.*degree, 0., 0.) * Translation3D(0., 0., -20.*cm);
   ring12_vol.setVisAttributes(desc.visAttributes(plm.attr<std::string>(_Unicode(vis_struc))));
 
+
+
+
+  //=============================
   // Inner supporting frame
+  //=============================
+
+  // // Version1: circle shape
+  // //
+  // Tube        Ssolid_ring12(8.5*cm, rmin, calo_module_length/2., phimin, phimax);
+  // Volume      Sring12_vol("Sring12", Ssolid_ring12, inner_ring_material);
+  // Sring12_vol.setVisAttributes(desc.visAttributes(plm.attr<std::string>(_Unicode(vis_struc))));
+  // Transform3D tr_global_Iring = RotationZYX(0., 0., 0.) * Translation3D(0., 0., 0.);
+
+
+  // // Version2: elliptical shape
+  // //
+  // EllipticalTube   inner_elliptical_support_a(9.*cm, 7.5*cm, calo_module_length/2.);
+  // EllipticalTube   inner_elliptical_support_b(8.5*cm, 7.*cm, calo_module_length/2.);
+  // SubtractionSolid inner_elliptical_support_substract(inner_elliptical_support_a, inner_elliptical_support_b, Position(0., 0., 0.));
+  // Volume           inner_elliptical_vol("inner_elliptical", inner_elliptical_support_substract, inner_ring_material);
+  // inner_elliptical_vol.setVisAttributes(desc.visAttributes(plm.attr<std::string>(_Unicode(vis_struc))));
+  // Transform3D tr_global_Iring_elli = RotationZYX(-15.*degree, 0., 0.) * Translation3D(2.05*cm, 0., 0.);
```

Comment:
Extensive commented sections should be removed. Since these sections are commented out, they are never compiled and when other parts of this file are changed they can and will become uncompilable or even wrong.

Reply:
got it, I remove those comment region in the last commit